File: t/CS/PDFFileDemo/GetPasswordDialog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace PDFFileDemo
{
   public partial class GetPasswordDialog : Form
   {
      private string _password;
      public string Password
      {
         get { return _password; }
      }

      public GetPasswordDialog()
      {
         InitializeComponent();
      }

      private void _showCharactersCheckBox_CheckedChanged(object sender, EventArgs e)
      {
         if(_showCharactersCheckBox.Checked)
         {
            _passwordTextBox.PasswordChar = (char)0;
         }
         else
         {
            _passwordTextBox.PasswordChar = '*';
         }
      }

      private void _okButton_Click(object sender, EventArgs e)
      {
         _password = _passwordTextBox.Text;
      }
   }
}

File: t/VB/PDFFileDemo/GetPasswordDialog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Namespace PDFFileDemo
   Partial Public Class GetPasswordDialog : Inherits Form
      Private _password As String
      Public ReadOnly Property Password() As String
         Get
            Return _password
         End Get
      End Property

      Public Sub New()
         InitializeComponent()
      End Sub

      Private Sub _showCharactersCheckBox_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles _showCharactersCheckBox.CheckedChanged
         If _showCharactersCheckBox.Checked Then
            _passwordTextBox.PasswordChar = ChrW(0)
         Else
            _passwordTextBox.PasswordChar = "*"c
         End If
      End Sub

      Private Sub _okButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _okButton.Click
         _password = _passwordTextBox.Text
      End Sub
   End Class
End Namespace
